web-app/HeimScripts/heatmap/run.R

```diff
@@ -176,3 +189,85 @@ add.subset.label <- function(df,label){
   }
   return(df)
 }
+
+computeDendrogram <- function(zScoreMatrix, distanceMeassure, linkageMethod) {
+    as.dendrogram(hclust(dist(zScoreMatrix, method=distanceMeassure), method=linkageMethod))
+}
```

Comment:
Not for now, but maybe there can be a (small?) improvement in computing time if the same distance matrix is not calculated multiple times... However, I think most of the time will be spent on the clustering itself though
For example here the same distance matrix is calculated 3 times:
  colDendrogramEuclideanComplete <- computeDendrogram(t(measurements), 'euclidean', 'complete')
+  colDendrogramEuclideanSingle <- computeDendrogram(t(measurements), 'euclidean', 'single')
+  colDendrogramEuclideanAverage <- computeDendrogram(t(measurements), 'euclidean', 'average')